src/test/mocks.ts

```diff
@@ -44,6 +58,16 @@ export class MockWatcher implements Watcher {
 	transfer(tx: Transfer): void {
 		this.txReceiptHandler(new TxReceipt(tx, new Account(0n, new Assets())));
 	}
+
+	phaseshift(bps: BalanceProofs) {
+		this.phaseShiftHandler();
```

Comment:
I think the `phaseShiftHandler` should be called after the proof handlers. Because strictly speaking, the proofs still belong to the old epoch and the phase shift is the last event before the new epoch starts.